static_frame/core/index_base.py

```diff
@@ -245,13 +245,13 @@ def _sample_and_key(self: I,
             count: int = 1,
             *,
             seed: tp.Optional[int] = None,
```

Comment:
At first I thought you weren't using `__future__.annotations`, but I see you. As such, how come you haven't opted for `int | None`?

Reply:
That syntax is only available in Python 3.9. I am still supporting Python 3.8, and it gets confusing to use some at runtime and some at hint time, so I default to old-style for now.